.github/workflows/solutions_ci.yml

```diff
@@ -0,0 +1,43 @@
+name: CI for SAF templates
+on:
+  pull_request:
+
+jobs:
+  check-solution-templates-changes:
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    outputs:
+      changed: ${{ steps.check.outputs.has_changes}}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Check for changes
+        id: check
+        run: |
+          CHANGES=$(git diff --name-only  ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} --  './src/ansys/templates/python/solution' './src/ansys/templates/python/osl_solution')
+          if [ -z "$CHANGES" ]; then
+            echo "has_changes=false" >> $GITHUB_OUTPUT
+            echo "No changes in the solution templates."
+          else
+            echo "has_changes=true" >> $GITHUB_OUTPUT
+            echo "Found changes in the solution templates."
+          fi
+
```

Comment:
I am wondering if all this logic is needed. Instead could this run be trigged by this?

```yaml
on:
  pull_request:
    branches: [ main ]
    paths:
      - 'your-directory-path/**
```

Usually your logic is needed since it operates in a single `ci.yaml` file, but here you are separated. So maybe could be simplified.

Reply:
@jacobevansgit I was not aware of that argument! Thanks for noticing, this is much better.